### PR TITLE
Implement assertions() in search #27

### PR DESCRIPTION
**Pull request by jazcap53**
… (#27)

Check name of optimizer member in SearchBase.
Check type of n_selection_iters member in RandomSearch.
Add self.assertions() line to __init__() in each of Searchbase, GridSearch, and RandomSearch.
Add call to super().assertions() in RandomSearch.

Author: jazcap53
Email: andrew.jarcho@gmail.com

### CODE REVIEW

File: pyswarms/utils/search/base_search.py [hunk 1]
```diff
@@ -13,6 +13,21 @@
...
+        ------
+        TypeError
+            When :code:`optimizer` is not named LocalBestPSO or
+                GlobalBestPSO.
+        """
+        # Check type of optimizer object
+        if self.optimizer.__name__ not in ('LocalBestPSO', 'GlobalBestPSO'):
+            raise TypeError('Parameter `optimizer` must be of type '
+                            'LocalBestPSO or GlobalBestPSO')
+
```
Comment (ljvmiranda921) [inline pyswarms/utils/search/base_search.py]: This is good for now and I can accept this in its current form. 😄 👍 

But I'm wondering how can we scale this when more search methods will be added. For example, Binary PSO is a single-objective method, and I think the same search methods are applicable to it. 

The same will go perhaps in multi-objective and constrained optimization. Since the search methods are "independent" whatever optimization method we are using. Meaning, it doesn't really care what kind of problem we are trying to solve, as long as we have an `optimize` and `optimizer` supplied as params. Then it's possible to apply the same search methods for these kind of techniques in the future. What do you think @SioKCronin?

I'm thinking of instead checking if the object passed in the `optimizer`  argument belongs to the `pyswarms.base` module. __OR__ check if the object passed in the `optimizer` method has an `optimize` attribute (thus using a `hasattr` as a checker). What do you think @jazcap53?

File: pyswarms/utils/search/base_search.py [hunk 2]
```diff
@@ -73,7 +90,7 @@ def generate_score(self, options):
         ----------
 
         options: dict
-            a dict of 5 hyperparameter values ('c1', 'c2', 'w', 'k', 'p' 
+            a dict of 5 hyperparameter values ('c1', 'c2', 'w', 'k', 'p'
```
Comment (ljvmiranda921) [inline pyswarms/utils/search/base_search.py]: Is a parenthesis missing here? 😄 ❓ 
I guess we can also rephrase this as:
```python
"""
a dict with the following keys: {'c1 ', 'c2', 'w', 'k', 'p'}
"""
```

File: pyswarms/utils/search/base_search.py (hunk 1 above)
Comment (jazcap53) [reply pyswarms/utils/search/base_search.py]: Hi @ljvmiranda921,
I'll of course make the changes you suggest with pleasure. :smile:
> I'd appreciate it if you can rebase your commit similar to the title of your pull request  

I had trouble getting git to accept a commit title with spaces in it. I'll try using `\<space>` instead of `<space>`, and if that doesn't work I'll try using `%20`.

> I'm wondering how can we scale this when more search methods will be added

I was wondering the same thing. I had trouble implementing it, though, since the `optimizer` parameter is a class rather than an instance -- I'll research it further and ask help if I can't find a solution.

> Is a parenthesis missing here?

Sure enough. I was afraid to change that line, though -- it's not mine. I just appears in the diff because my vim is set to eliminate trailing spaces on save of a .py file. I'll rephrase it according to your suggestion.

I'm enjoying this very much...  :+1:

File: pyswarms/utils/search/base_search.py (hunk 2 above)
Comment (jazcap53) [reply pyswarms/utils/search/base_search.py]: Hi @ljvmiranda921 

> Is a parenthesis missing here?

Sure enough. :smile: I was afraid to change that line, though -- it's not mine. I just appears in the diff because my vim is set to eliminate trailing spaces on save of a .py file. I'll rephrase it according to your suggestion.

File: pyswarms/utils/search/base_search.py (hunk 1 above)
Comment (jazcap53) [reply pyswarms/utils/search/base_search.py]: @ljvmiranda921   Figured out that I have to use `issublcass()`. Simple (embarrassing). :smile:

Comment (jazcap53) [reply pyswarms/utils/search/base_search.py]: @ljvmiranda921    o.i.c.   `if not hasattr(self.optimizer, 'optimize'):`
I'll use that unless you would prefer something else  :smile:

**Comment by jazcap53** [comment]
Hi @ljvmiranda921 !

I've followed all instructions, I hope correctly. Google and Stack Overflow have been getting a lot of my business lately! :smile:

If there's anything else I need to do, please of course let me know. 

Next I plan to study up on Travis CI and write some tests for the 'Go the extra mile?' portion of the Issue.

:+1:

**Comment by ljvmiranda921** [comment]
Hi @jazcap53 ! All is good! I will merge this now 👍 

Congratulations for your first contribution!!! :D 

For the extra mile, no need to actually study travis-ci because I have already set that one up beforehand. You just need to write the unit tests, and travis-ci will automatically run your tests using `discover`. 😄 

So in case that one of the tests didn't pass, it will send a notification here in Github and also in my e-mail. In the 'Extra mile', you will just write the unit tests that travis-ci will run. 😄

**Comment by jazcap53** [comment]
@ljvmiranda921 
Yesssssss! :smile: :+1:
Thanks so much for all your help. Looking forward to doing the tests!
